features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/RoomListPresenter.kt

```diff
@@ -188,6 +192,27 @@ class RoomListPresenter @Inject constructor(
         matrixUser.value = client.getCurrentUser()
     }
 
+    private fun CoroutineScope.showContextMenu(event: RoomListEvents.ShowContextMenu, contextMenuState: MutableState<RoomListState.ContextMenu>) = launch {
+        val initialState = RoomListState.ContextMenu.Shown(
+            roomId = event.roomListRoomSummary.roomId,
+            roomName = event.roomListRoomSummary.name,
+            isDm = event.roomListRoomSummary.isDm,
+            isFavorite = event.roomListRoomSummary.isFavorite,
+            markAsUnreadFeatureFlagEnabled = featureFlagService.isFeatureEnabled(FeatureFlags.MarkAsUnread),
+            hasNewContent = event.roomListRoomSummary.hasNewContent
+        )
+        contextMenuState.value = initialState
+        client.getRoom(event.roomListRoomSummary.roomId)?.use { room ->
```

Comment:
Let me know if you want me to take a look at it while you're (in theory 😅 ) away.

Reply:
I've pushed the changes, let me know!

Reply:
It feels a bit weird the `takeWhile` is after the `onEach`,  but it works well enough and it's simpler than using `combine()` and passing an extra parameter everywhere. LGTM!

Reply:
Actually I think it should be before `onEach` you're right, I'm changing it.

Reply:
Yeah I don't know if it's better or worse actually, but I guess the result is almost the same.

Reply:
This is how it looks with `combine` instead

```
val isContextMenuShownFlow = snapshotFlow { contextMenuState.value is RoomListState.ContextMenu.Shown }
                .distinctUntilChanged()

            val isFavoriteFlow = room.roomInfoFlow
                .map { it.isFavorite }
                .distinctUntilChanged()

            combine(isContextMenuShownFlow, isFavoriteFlow) { isContextMenuShown, isFavorite ->
                isContextMenuShown to isFavorite
            }
                .takeWhile { (isShowingContextMenu, _) -> isShowingContextMenu }
                .onEach { (_, isFavorite) ->
                    contextMenuState.value = initialState.copy(
                        isFavorite = isFavorite,
                    )
                }.collect()
```

Reply:
This is what I had tested locally. The previous version seems simpler, right? It's a bit harder to understand why it's after the onEach, but as long as we have at least 1 item emitted by the room info flow it should be fine.

Reply:
If you put it before it doesn't work because it won't cancel the stream because of the flatMapLatest creating a new stream.

Also the statement "as long as we have at least 1 item emitted by the room info flow it should be fine" is true also for the combine operator